caskdata/cdap
# Feature/metric store integrated

This is an aggregated PR of the MetricStore. It was done as series of PRs and all pieces were reviewed.

* initial metric store integration #1251 MERGED
* metrics discovery #1267 MERGED
* unit-test framework #1266 - MERGED
* queue length #1268 - MERGED
* delete metrics #1272 - MERGED
* ttl for non-hbase metric datasets #1272 - MERGED
* test on distributed cdap #1274 - MERGED
* interpolators #1289 MERGED
* remove + .v2 in dataset names #1283 - MERGED
* upgrade metric datasets (part, see CDAP-1458 for rest) #1283 MERGED
* reserve byte for format evolution in FactTable #1284 MERGED
* all unit-tests to pass #1298 MERGED #1299 MERGED
* support limit in cube/fact table - DEFERRED to after merge of this branch
* metrics.data package - move - DEFERRED to after merge of this branch
